### ISSUE TITLE
How to use Porcupine with Pyenv?

### ISSUE THREAD

**Issue by FilipeMarch** [Akuli/porcupine#154]
I am using [Pyenv](https://realpython.com/intro-to-pyenv/) to create virtual environments. 
Before running my python files, I first do this:
```pyenv activate my_virtual_environment```
Then I proceed to run my python files with:
```python my_python_file.py```

How to reproduce the bug in Porcupine:

1. Create a virtual environment using Pyenv:
   ```pyenv virtualenv 3.8 porcupine_venv``` 
2. Create a script.py file with Porcupine.
3. Paste this code into it:

```python
import os
os.system('pyenv init -')
os.system('pyenv virtualenv-init -')
os.system("pyenv activate porcupine_venv")
```

4. Run it with Porcupine.

You should see this message:

```shell
export PATH="/home/user/.pyenv/plugins/pyenv-virtualenv/shims:${PATH}";
export PYENV_VIRTUALENV_INIT=1;
_pyenv_virtualenv_hook() {
  local ret=$?
  if [ -n "$VIRTUAL_ENV" ]; then
    eval "$(pyenv sh-activate --quiet || pyenv sh-deactivate --quiet || true)" || true
  else
    eval "$(pyenv sh-activate --quiet || true)" || true
  fi
  return $ret
};
if ! [[ "$PROMPT_COMMAND" =~ _pyenv_virtualenv_hook ]]; then
  PROMPT_COMMAND="_pyenv_virtualenv_hook;$PROMPT_COMMAND";
fi

Failed to activate virtualenv.
Perhaps pyenv-virtualenv has not been loaded into your shell properly.
Please restart current shell and try again.
```

To avoid getting into compatibility issues, I use virtual environments to isolate each project. I don't know how to integrate Porcupine with Pyenv. I googled it but haven't found a similar issue.

**Comment by Akuli** [comment]
Running files inside porcupine was created before I was aware of virtualenvs, which is why I never use it, lol. If you really want to run files inside Porcupine, you can change the running command in `filetypes.toml` to use the virtualenv's python.

**Comment by Akuli** [comment]
For whatever reason, the first google result for "porcupine filetypes.toml" is Porcupine's source code, not documentation. The documentation is here: https://github.com/Akuli/porcupine/wiki/Getting-Porcupine-to-work-with-a-programming-language

**Comment by Akuli** [comment]
Seems like I misunderstood what pyenv is. It's not a thing for using virtualenvs, but for using different Python versions.

Looking at [this thing](https://realpython.com/intro-to-pyenv/), it seems like you should put something like this to your `filetypes.toml`:

```toml
[Python]
commands = {run = "/home/username/.pyenv/shims/python {file}"}
```

Does that help?

**Comment by Akuli** [comment]
Somehow even this doesn't work (not on my computer, confirmed that python.exe exists in that path):

```toml
[Python]
commands = {run = 'C:\Users\Martin\tetris\env\Scripts\python.exe {file}'}
```

I think I'll try to figure this out tomorrow

**Comment by Akuli** [comment]
Now that should work. Fixed in 824411b

**Comment by Akuli** [comment]
Porcupine now has much better venv support than before:
- If your project contains a venv, it shows up with a yellow `venv` text and Porcupine uses it automatically when you run commands.
- If your project contains multiple venvs, you can right-click them to choose which one to use.
